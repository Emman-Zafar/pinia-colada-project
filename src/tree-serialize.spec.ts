import { describe, expect, it } from 'vitest'
import { TreeMapNode } from './tree-map'
import type { UseQueryEntry } from './query-store'
import { createQueryEntry, createTreeMap, serialize } from './query-store'

describe('tree-map serialization', () => {
  it('basic serialization', () => {
    const tree = new TreeMapNode<UseQueryEntry>()
    tree.set(['a'], createQueryEntry(['a'], 'a'))
    tree.set(['a', 'b'], createQueryEntry(['a', 'b'], 'ab'))
    tree.set(['a', 'b', 'c'], createQueryEntry(['a', 'b', 'c'], 'abc'))
    tree.set(['d', 'e', 'f'], createQueryEntry(['d', 'e', 'f'], 'def'))
    expect(serialize(tree)).toMatchSnapshot()
    expect(serialize(tree)).toEqual(serialize(createTreeMap(serialize(tree))))
  })

  it('works with empty', () => {
    const tree = new TreeMapNode<UseQueryEntry>()
    expect(serialize(tree)).toEqual([])
    expect(serialize(tree)).toEqual(serialize(createTreeMap()))
    expect(serialize(tree)).toEqual(serialize(createTreeMap([])))
  })
})
