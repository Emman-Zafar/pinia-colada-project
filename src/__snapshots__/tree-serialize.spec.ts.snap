// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tree-map serialization > basic serialization 1`] = `
[
  [
    "a",
    [
      "a",
      null,
      0,
    ],
    [
      [
        "b",
        [
          "ab",
          null,
          0,
        ],
        [
          [
            "c",
            [
              "abc",
              null,
              0,
            ],
            undefined,
          ],
        ],
      ],
    ],
  ],
  [
    "d",
    undefined,
    [
      [
        "e",
        undefined,
        [
          [
            "f",
            [
              "def",
              null,
              0,
            ],
            undefined,
          ],
        ],
      ],
    ],
  ],
]
`;
